platform/parser/api/src/main/java/org/codice/ddf/parser/Parser.java

```diff
@@ -41,24 +46,52 @@
      *
      * @param configurator object containing the relevant configuration information needed to
      *                     perform the conversion
-     * @param obj the root of the object graph to convert
-     * @param os the output stream on which the converted object is written
+     * @param obj          the root of the object graph to convert
+     * @param os           the output stream on which the converted object is written
      * @throws ParserException
      */
     void marshal(ParserConfigurator configurator, Object obj, OutputStream os)
             throws ParserException;
 
     /**
+     * @param configurator
+     * @param obj
+     * @param node
+     * @throws ParserException
+     */
+    void marshal(ParserConfigurator configurator, Object obj, Node node) throws ParserException;
+
+    /**
      * Converts a representation of an object graph into an instance of type {@code T}.
      *
      * @param configurator object containing the relevant configuration information needed to
      *                     perform the conversion
-     * @param cls {@code Class} for the conversion
-     * @param stream input stream which is read for the object data
-     * @param <T> expected return object
+     * @param cls          {@code Class} for the conversion
+     * @param stream       input stream which is read for the object data
+     * @param <T>          expected return object
      * @return an object of type {@code T} as read and converted from the stream
      * @throws ParserException
      */
     <T> T unmarshal(ParserConfigurator configurator, Class<? extends T> cls, InputStream stream)
             throws ParserException;
+
+    /**
+     * @param <T>
+     * @param cls
+     * @param node
+     * @return an object of type {@code T} as read and converted from the stream
+     * @throws ParserException
+     */
+    <T> JAXBElement<T> unmarshal(ParserConfigurator configurator, Class<? extends T> cls, Node node)
```

Comment:
I'm assuming that the callers need the `JAXBElement`s for their own purposes. If not, then this should be changed to return `T`. It isn't terrible if it stays like this _if it needs to be like this_ as this has become tightly bound to XML/JAXB at this point.